lib/gpu_memory_service/tests/test_gms_write_publication.py

```
@@ -0,0 +1,624 @@
+    for local_success in (True, False):
+        events = []
+        all_events.append(events)
+        prepared = common_utils.prepare_gms_write(_FakeAllocator(events), object())
+        model_loader._store_pending_gms_write(prepared)
+
+        def coordinate(success):
+            coordination_inputs.append(success)
+            return False
+
+        def profile():
+            if not local_success:
```

Ruff (0.15.20)

[warning] Function definition does not bind loop variable `local_success`

(B023)